src/fat_beagle.cpp

```diff
@@ -68,38 +70,33 @@ double FatBeagle::LogLikelihoodInternals(
   return log_like;
 }
 
-double FatBeagle::LogLikelihood(const UnrootedTree &tree) const {
+double FatBeagle::LogLikelihood(const UnrootedTree &tree, std::optional<PhyloFlags> flags /*= std::nullopt*/) const {
   auto detrifurcated_tree = tree.Detrifurcate();
   return LogLikelihoodInternals(detrifurcated_tree.Topology(),
                                 detrifurcated_tree.BranchLengths());
 }
 
-double FatBeagle::UnrootedLogLikelihood(const RootedTree &tree) const {
+double FatBeagle::UnrootedLogLikelihood(const RootedTree &tree, std::optional<PhyloFlags> flags /*= std::nullopt*/) const {
   return LogLikelihoodInternals(tree.Topology(), tree.BranchLengths());
 }
 
-double LogDeterminantJacobian(const RootedTree &tree) {
-  double log_det_jacobian = 0.0;
-  size_t leaf_count = tree.LeafCount();
-  tree.Topology()->TripleIdPreorderBifurcating(
-      [&log_det_jacobian, &tree, leaf_count](int node_id, int sister_id,
-                                             int parent_id) {
-        if (node_id >= leaf_count) {
-          log_det_jacobian +=
-              std::log(tree.node_heights_[parent_id] - tree.node_bounds_[node_id]);
-        }
-      });
-  return log_det_jacobian;
-}
-
-double FatBeagle::LogLikelihood(const RootedTree &tree) const {
+double FatBeagle::LogLikelihood(
+    const RootedTree &tree,
+    const std::optional<PhyloFlags> flags /*= std::nullopt*/) const {
+  double log_likelihood = 0.0f;
   std::vector<double> branch_lengths = tree.BranchLengths();
   const std::vector<double> &rates = tree.GetRates();
   for (size_t i = 0; i < tree.BranchLengths().size() - 1; i++) {
     branch_lengths[i] *= rates[i];
   }
-  return LogLikelihoodInternals(tree.Topology(), branch_lengths) +
-         LogDeterminantJacobian(tree);
+  log_likelihood += LogLikelihoodInternals(tree.Topology(), branch_lengths);
+  // Only exclude log_det_jacobian if flagged specifically.
+  if (PhyloFlags::IsFlagNotChecked(
+          flags,
+          PhyloFlags::loglikelihood_options_.exclude_log_det_jacobian_likelihood_)) {
+    log_likelihood += ::LogDetJacobianHeightTransform(tree);
```

Comment:
Do we need this `::`?

Reply:
No. Initially, there was a static function with the same name, so it was temporarily needed.